src/core/server.cc

```diff
@@ -415,6 +416,12 @@ InferenceServer::SharedMemoryAddress(
       name, offset, byte_size, shm_mapped_addr);
 }
 
+Status
+InferenceServer::GetSharedMemoryStatus(SharedMemoryControlResponse* shm_status)
+{
+  return shared_memory_manager_->GetSharedMemoryStatus(shm_status);
```

Comment:
Need to check ready_state_ and update inflight_request_counter_ as in UnregisterAllSharedMemory.